### PR TITLE
Meta cloning

### PR DESCRIPTION
**Pull request by jeroendesloovere**
## Type

- Feature

## Pull request description

> Allows us to create "clone/copy" methods on other entities and duplicate the meta.
I've used this in a "Categories module" "CopyCategoriesToOtherLocaleHandler".

Behavior:
* $id is set to null, so when the command bus executes everything a new meta id will be given
* $url creates an unique url

### CODE REVIEW

File: src/Common/Doctrine/Entity/Meta.php [hunk 1]
```diff
@@ -167,6 +168,15 @@ public function update(
         $this->seoIndex = $seoIndex;
     }
 
+    public function duplicate()
```
Comment (carakas) [inline src/Common/Doctrine/Entity/Meta.php]: https://secure.php.net/manual/en/language.oop5.cloning.php

**Comment by jeroendesloovere** [comment]
Can come in handy for this issue https://github.com/forkcms/forkcms/issues/2092

File: src/Common/Doctrine/Entity/Meta.php [hunk 2]
```diff
@@ -170,11 +170,8 @@ public function update(
 
     public function __clone()
```
Comment (carakas) [inline src/Common/Doctrine/Entity/Meta.php]: can you add the :void typehint
did you also test it?

Comment (jeroendesloovere) [reply src/Common/Doctrine/Entity/Meta.php]: - : void return type declaration is not allowed
- nope, didn't tested this

Comment (carakas) [reply src/Common/Doctrine/Entity/Meta.php]: can you maybe test this by writing a test for it :)

**Comment by carakas** [comment]
@jeroendesloovere features do not belong in bugfix releases :) http://semver.org/

**Comment by jeroendesloovere** [comment]
@carakas test added. Everything is working fine.

File: src/Common/Doctrine/Entity/Meta.php [hunk 3]
```diff
@@ -167,6 +168,12 @@ public function update(
         $this->seoIndex = $seoIndex;
     }
 
+    public function __clone()
+    {
+        $this->id = null;
+        $this->url = Model::addNumber($this->url);
```
Comment (carakas) [inline src/Common/Doctrine/Entity/Meta.php]: hmm we can't be sure that this url will be unique
if you have my-url and my-url-2 and you clone my-url that will also turn in to my-url-2

**Comment by carakas** [comment]
This is probably not something you can solve like this so I'm going to close this one

**Pull request closed without merging**